compose/project.py

```diff
@@ -326,6 +329,30 @@ def _inject_deps(self, acc, service):
         dep_services.append(service)
         return acc + dep_services
 
+    def _parallel_executor(self, command, containers, doing_msg, done_msg, **options):
+        """
+        Execute a given command upon a list of containers in parallel.
+        """
+        max_workers = os.environ.get('MAX_WORKERS', DEFAULT_MAX_WORKERS)
+
+        def container_command_execute(container, command, **options):
+            log.info("{} {}...".format(doing_msg, container.name))
+            return getattr(container, command)(**options)
+
+        with concurrent.futures.ThreadPoolExecutor(max_workers=max_workers) as executor:
+            future_container = {
+                executor.submit(
+                    container_command_execute,
+                    container,
+                    command,
+                    **options
+                ): container for container in containers
+            }
+
+            for future in concurrent.futures.as_completed(future_container):
+                container = future_container[future]
+                log.info("{} {}".format(done_msg, container.name))
+
```

Comment:
Looks great. Since this doesn't use `self` any more, could it be moved into `compose.utils`?